### PR TITLE
JSON-like constructors

### PR DESCRIPTION
**Pull request by SimonCockx**
See [issue](https://github.com/REGnosys/rosetta-dsl/issues/398) for details.

## Type of change

- New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: rosetta-ide/.classpath [hunk 1]
```diff
@@ -62,5 +62,11 @@
 			<attribute name="test" value="true"/>
 		</attributes>
 	</classpathentry>
+	<classpathentry excluding="**" kind="src" output="target/test-classes" path="src/test/resources">
```
Comment (hugohills-regnosys) [inline rosetta-ide/.classpath:65]: Is this change expected?

File: rosetta-runtime/.classpath [hunk 2]
```diff
@@ -67,5 +67,11 @@
 			<attribute name="test" value="true"/>
 		</attributes>
 	</classpathentry>
+	<classpathentry excluding="**" kind="src" output="target/test-classes" path="src/test/resources">
```
Comment (hugohills-regnosys) [inline rosetta-runtime/.classpath:70]: Is this expected

File: documentation/rosetta-modelling-component.md [hunk 3]
```diff
@@ -662,6 +663,66 @@ E.g.:
...
+  favouriteAnimal: "bear",
+  mentor: Employee {
+    name: "Michael Scott",
+    age: empty,
+    favouriteAnimal: empty,
+    mentor: empty
+  }
+}
+```
+
```
Comment (lolabeis) [inline documentation/rosetta-modelling-component.md:699]: I would add the alternative way of creating the "Dwight Schrute" record, which is first to set "Michael Scott" as an alias, and then use that alias when constructing Dwight.

I also understand that you can use Rosetta expressions within the instantiation syntax, for instance a path extraction, so you could add an example accordingly.

Those 2 points are described in the issue.

File: documentation/rosetta-modelling-component.md [hunk 4]
```diff
@@ -662,6 +663,66 @@ E.g.:
...
+```
+<type name> {
+  <field name 1>: <field value 1>,
+  <field name 2>: <field value 2>,
+  etc
+}
+```
+
+As an example, consider the following data type representing an employee of a company.
+``` Haskell
```
Comment (lolabeis) [inline documentation/rosetta-modelling-component.md:678]: Our "theme" for examples throughout the documentation is related to vehicles. I think we should continue to follow the same convention. And accordingly, those examples could be added to the demo model.

File: documentation/rosetta-modelling-component.md [hunk 5]
```diff
@@ -662,6 +663,66 @@ E.g.:
...
+    age: empty,
+    favouriteAnimal: empty,
+    mentor: empty
+  }
+}
+```
+
+#### Triple-Dot Syntax
+
+Notice that in the example above most fields of Michael Scott are actually `empty`. Types with many optional attributes is common practice in Rosetta models such as the CDM, and assigning `empty` to them explicitly can be verbose. We can solve this by using the triple-dot keyword `...`, which will implicitly assign `empty` to all absent attributes.
```
Comment (lolabeis) [inline documentation/rosetta-modelling-component.md:702]: Typo in the second sentence, not syntactically correct. "Having" missing perhaps?

**Review by davidalk — APPROVED** [review]

**Pull request merged**